File: Cuarto/Laboratorio/Talleres/RobocodeLabo/src/viejo2/Robotito.java
package viejo2;

import robocode.JuniorRobot;

public class Robotito extends JuniorRobot {

    private Estratega strategist = TwoPhaseCombatStrategy.INSTANCE;

    private CombatStrategy strategy;

    @Override
    public void run() {
        setStrategy(strategist.elegirEstrategia(this));
        while(true){
            onTick();
        }
    }

    public void setStrategy(CombatStrategy newStrat) {
        strategy = newStrat;
    }
    private void onTick(){
        strategist.elegirEstrategia(this).onTick();
    }

    public void onHitWall(){
      strategist.elegirEstrategia(this).onHitWall(); }

    public void onHitRobot() { strategist.elegirEstrategia(this).onHitRobot(); }

    public void onHitByBullet() {
      strategist.elegirEstrategia(this).onHitByBullet();
    }

    public void onScannedRobot(){ strategist.elegirEstrategia(this).onScannedRobot(); }
}

File: Cuarto/Laboratorio/Talleres/RobocodeLabo/src/robotitoKotlin/Robotito.kt
package robotitoKotlin

import robocode.JuniorRobot

class Robotito: JuniorRobot() {
  private var strategist = TwoPhaseCombatStrategy.INSTANCE

  init{
    TwoPhaseCombatStrategy.robot = this
  }
  override fun run() {
    while (true){
        onTick()
    }
  }

  private fun onTick() =  strategist.elegirEstrategia().onTick()

  override fun onHitWall() = strategist.elegirEstrategia().onHitWall()

  override fun onHitRobot() = strategist.elegirEstrategia().onHitRobot()

  override fun onHitByBullet() = strategist.elegirEstrategia().onHitByBullet()

  override fun onScannedRobot() = strategist.elegirEstrategia().onScannedRobot()




}